x-pack/plugins/actions/server/action_type_registry.ts

```diff
@@ -58,6 +61,19 @@ export class ActionTypeRegistry {
     );
   }
 
+  /**
+   * Returns true if action type is enabled or it is a preconfigured action type.
+   */
+  public isActionTypeExecutable(id: string) {
+    return (
+      this.isActionTypeEnabled(id) ||
+      (!this.isActionTypeEnabled(id) &&
+        this.preconfiguredActions.find(
+          preconfiguredAction => preconfiguredAction.actionTypeId === id
+        ) !== undefined)
+    );
+  }
```

Comment:
Looking at this logic and I think we need to compare `id` and `actionTypeId` within the `preconfiguredAction` find function. It could happen that two connectors have the same id of different type or this could allow the execution of any action of a given type that is preconfigured.

With that, the function name could be renamed to `isActionExecutable` and take an action id + action type id as arguments (`isActionTypeExecutable(actionId, actionTypeId)`)